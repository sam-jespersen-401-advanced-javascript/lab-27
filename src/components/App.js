import React from 'react';
import Header from './Header';
import Footer from './Footer';
import Character from './Character';
import characters from '../../public/data';

const chars = characters.map(char => {
  return <Character
    key={char.name}
    name={char.name}
    img={char.img}
    species={char.species}
    quote={char.quote}
  />;
});


const App = () => (
  <>

    <Header />
    <div style={{ display: 'flex', flexWrap: 'wrap' }}>
      {[...chars]}
    </div>
    <Footer />

  </>
);

export default App;
