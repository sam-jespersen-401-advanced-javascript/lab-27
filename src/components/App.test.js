import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('app component', () => {
  it('renders an app', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
