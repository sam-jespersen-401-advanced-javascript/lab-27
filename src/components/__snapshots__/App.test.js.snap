// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders an app 1`] = `
<Fragment>
  <Header />
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
      }
    }
  >
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
      key="Bender"
      name="Bender"
      quote="Bite my shiny metal ass."
      species="Robot"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904145/leela.png"
      key="Leela"
      name="Leela"
      quote="...at least here you'll be treated with dignity. Now strip naked and get on
the probulator."
      species="Mutant"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904133/fry.png"
      key="Fry"
      name="Fry"
      quote="...it's what seperates humans and robots from animals and animal robots."
      species="Human"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904047/calculon.jpg"
      key="Calculon"
      name="Calculon"
      quote="Me too. I'm filled with a large number of powerful emotions."
      species="Robot"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/farnsworth.png"
      key="Professor Farnsworth"
      name="Professor Farnsworth"
      quote="...each pound of which weighs over 10,000 pounds."
      species="Human"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904049/Zapp_Brannigan.png"
      key="Zapp Brannigan"
      name="Zapp Brannigan"
      quote="In the game of chess you can never let your adversary see your pieces."
      species="Human"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/Dr_John_Zoidberg.png"
      key="Dr Zoidberg"
      name="Dr Zoidberg"
      quote="Scalpel... blood bucket... priest. Next patient!"
      species="Alien"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904013/amy.png"
      key="Amy"
      name="Amy"
      quote="If I ever feel lonely, I'll just look over at this disfiguring scar and think
of you."
      species="Human"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904013/Hermes.png"
      key="Hermes"
      name="Hermes"
      quote="Kudos, Bender! You got mangled and became a singer. Both our dreams came true."
      species="Human"
    />
    <Character
      img="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/Kif.png"
      key="Kif"
      name="Kif"
      quote="It's not uncharted, you lost the chart!"
      species="Alien"
    />
  </div>
  <Footer />
</Fragment>
`;
